### ISSUE TITLE
Internal error during login, jwt.split() with jwt value None

### ISSUE THREAD

**Issue by karlcz** [informatics-isi-edu/webauthn#57]
On newer Fedora-based VMs we are seeing frequent but sporadic failures during login using globus_auth providers.

A typical stack-trace looks like:
```
'got exception "\\'NoneType\\' object has no attribute \\'split\\'" during _db_wrapper()'
['Traceback (most recent call last):\\n',
 '  File "/usr/lib/python2.7/site-packages/webauthn2/util.py", line 519, in _db_wrapper\\n    val = db_thunk(db)\\n',
 '  File "/usr/lib/python2.7/site-packages/webauthn2/rest.py", line 458, in db_body\\n    self.context.client = self.manager.clients.login.login(self.manager, self.context, db, **storage)\\n',
 '  File "/usr/lib/python2.7/site-packages/webauthn2/providers/globus_auth.py", line 50, in login\\n    user_id = oauth2.OAuth2Login.login(self, manager, context, db, **kwargs)\\n',
 '  File "/usr/lib/python2.7/site-packages/webauthn2/providers/oauth2.py", line 308, in login\\n    id_result=self.verify_signed_jwt_with_keys(raw_id_token, keys, self.provider.cfg.get(\\'client_id\\'))\\n',
 '  File "/usr/lib/python2.7/site-packages/webauthn2/providers/oauth2.py", line 457, in verify_signed_jwt_with_keys\\n    segments = jwt.split(\\'.\\')\\n',
 "AttributeError: 'NoneType' object has no attribute 'split'\\n"]

Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/web/application.py", line 239, in process
    return self.handle()
  File "/usr/lib/python2.7/site-packages/web/application.py", line 230, in handle
    return self._delegate(fn, self.fvars, args)
  File "/usr/lib/python2.7/site-packages/web/application.py", line 447, in _delegate
    return handle_class(f)
  File "/usr/lib/python2.7/site-packages/web/application.py", line 438, in handle_class
    return tocall(*args)
  File "/usr/lib/python2.7/site-packages/webauthn2/rest.py", line 135, in wrapper
    return original_method(*args)
  File "/usr/lib/python2.7/site-packages/webauthn2/rest.py", line 320, in GET
    return self._login_get_or_post(web.input())
  File "/usr/lib/python2.7/site-packages/webauthn2/rest.py", line 475, in _login_get_or_post
    result = self._db_wrapper(db_body)
  File "/usr/lib/python2.7/site-packages/webauthn2/util.py", line 575, in _db_wrapper
    raise ev
AttributeError: 'NoneType' object has no attribute 'split'
```

**Comment by ljpearlman** [comment]
Okay, I'm pretty sure at this point that this is a manifestation of https://github.com/pyca/cryptography/issues/3282

One thing that makes me sure of this is that I've run some tests and found the error messages referenced by that bug on Fedora systems when logins are failing but not when they're succeeding.

This leaves us with two options: try to upgrade to a fixed version of the Python cryptography module, or work around it by using a single global instance of mod_wsgi. @karlcz, what do you think?

**Comment by ljpearlman** [comment]
Note: those messages are in `error_log`, not `ssl_error_log`. And they're not timestamped, so you have to watch the log file while you're attempting to log in.

**Comment by karlcz** [comment]
We determined that we did not previously use the Fedora RPM for python-cryptography but it was getting pulled in on new Fedora 25 builds and interfering with our stack due to that bug in the 1.5 version currently shipped in Fedora.

A temporary workaround is to forcibly `dnf remove python2-cryptography` from the system before building our stack, so that a different set of third-party libraries are chosen for the pip dependencies in the globus_auth and oauth2 packages.  Once they start shipping version 1.7 of the cryptography package, this may no longer be necessary.
